Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -791,20 +820,95 @@ public extension ChatChannelController {
             channelModificationFailed(completion)
             return
         }
-
-        guard let messageId = messageId ?? messages.first?.id else {
+        
+        guard let messageId = messageId ?? lastNewestMessageId ?? messages.first?.id else {
             log.error(ClientError.ChannelEmptyMessages().localizedDescription)
             callback { completion?(ClientError.ChannelEmptyMessages()) }
             return
         }
 
-        channelQuery.pagination = MessagesPagination(pageSize: limit, parameter: .greaterThan(messageId))
+        guard !hasLoadedAllNextMessages && !isLoadingNextMessages else {
+            return
+        }
 
+        isLoadingNextMessages = true
+        channelQuery.pagination = MessagesPagination(pageSize: limit, parameter: .greaterThan(messageId))
         updater.update(channelQuery: channelQuery, isInRecoveryMode: false, completion: { result in
-            self.callback { completion?(result.error) }
+            self.isLoadingNextMessages = false
+            switch result {
+            case let .success(payload):
+                self.updateNewestFetchedMessageId(with: payload)
+                self.callback { completion?(nil) }
+            case let .failure(error):
+                self.callback { completion?(error) }
+            }
         })
     }
 
+    /// Load messages around the given message id. Useful to jump to a message which is not loaded yet.
+    ///
+    /// Cleans the current messages of the channel and loads the message with the given id,
+    /// and the messages around it depending on the limit provided.
+    ///
+    /// Ex: If the limit is 25, it will load the message and 12 on top and 12 below it. (25 total)
```

Comment:
If it is 1, it will only load the message that is passed as an argument (assuming the server is working this way). Then, it should still work, since that message will be used as the cursor for loading the next page, and the previous page as well. In terms of the test, we can verify this last behaviour, since it is the only difference